src/server/routes/parcelDelivery.routes.spec.ts

```diff
@@ -175,4 +181,73 @@ describe('Parcel delivery route', () => {
       data_base64: SERVICE_MESSAGE_CONTENT.toString('base64'),
     });
   });
+
+  describe('Incoming PDA', () => {
+    const pdaContentType = 'application/vnd+relaycorp.awala.pda-path';
+
+    test('Valid PDA should be stored', async () => {
+      const certificatePath = await generatePDACertificationPath(KEY_PAIR_SET);
+      const pdaPath = new CertificationPath(certificatePath.pdaGrantee, [
+        certificatePath.privateEndpoint,
+        certificatePath.privateGateway,
+      ]);
+      const peerConnectionParams = new PrivateEndpointConnParams(
+        PEER_KEY_PAIR.privateGateway.publicKey,
+        PEER_ADDRESS,
+        pdaPath,
+      );
+      const { parcelSerialized } = await generateParcel(
+        parcelRecipient,
+        KEY_PAIR_SET,
+        new Date(),
+        sessionKey,
+        pdaContentType,
+        Buffer.from(await peerConnectionParams.serialize()),
+      );
+      const spyOnSavePeerEndpointChannel = jest.spyOn(
+        server.activeEndpoint,
+        'savePeerEndpointChannel',
+      );
+
+      const response = await server.inject({
+        ...validRequestOptions,
+        payload: parcelSerialized,
+      });
+
+      expect(spyOnSavePeerEndpointChannel).toHaveBeenCalledOnceWith(
+        expect.objectContaining({
+          internetGatewayAddress: peerConnectionParams.internetGatewayAddress,
+        }),
+        server.mongoose,
+      );
+      expect(response).toHaveProperty('statusCode', HTTP_STATUS_CODES.ACCEPTED);
+      expect(logs).toContainEqual(partialPinoLog('info', 'Peer connection params stored'));
+    });
+
+    test('Invalid connection params should be accepted but not stored', async () => {
+      const { parcelSerialized } = await generateParcel(
+        parcelRecipient,
+        KEY_PAIR_SET,
+        new Date(),
+        sessionKey,
+        pdaContentType,
+        Buffer.from('Invalid PDA'),
```

Comment:
This test is now testing the malformed scenario. Please:

- Update the title of this test accordingly.
- Restore the test for the well-formed yet invalid scenario.

Also:

```suggestion
Buffer.from('Malformed PDA'),
````